hs_labels/models.py

```diff
@@ -0,0 +1,876 @@
+        label_resource = this_resource.rlabels
+
+        if not isinstance(this_label, basestring):
+            raise HSLUsageException("Label is not text")
+
+        label_string = UserLabels.clean_label(this_label)                 # remove leading and trailing spaces
+
+        UserResourceLabels.objects.filter(rlabels=label_resource,
+                                          label__exact=label_string,
+                                          kind = LabelCodes.LABEL,
+                                          ulabels=self).delete()
+
+    def clear_resource_labels(self, this_resource):
+        """ Clear all labels for a resource
+
+        :param this_resource: BaseResource for which to clear all labels
+        :return: none.
+        """
+        # check for user error
+        if not isinstance(this_resource, BaseResource):
+            raise HSLUsageException("Target is not a resource")
+        label_resource = this_resource.rlabels
+        UserResourceLabels.objects.filter(rlabels=label_resource,
+                                          ulabels=self,
+                                          kind=LabelCodes.LABEL).delete()
+
+    ##########################################
+    # filed resources
+    ##########################################
+
+    @staticmethod
+    def clean_folder(name):
+        folder_string = re.sub('/+', r'/', name)         # no doubled //
+        folder_string = re.sub(r'^(/|\s)+', r'', folder_string)   # no leading /
+        folder_string = re.sub(r'(/|\s)+$', r'', folder_string)   # no trailing /
+        folder_string = re.sub(r'\s+', r' ', folder_string)       # collapse multiple whitespace, including tabs
+        folder_string = re.sub(r'\s*/\s*', r'/', folder_string)   # no whitespace around /'s
+        return folder_string
+
+    def file_resource(self, this_resource, this_folder):
+        """
+        Specify a file folder for a resource.
+
+        :param this_resource: Resource to share.
+        :param this_folder: Folder name to assign to resource
+        :return: None
+
+        Users are allowed to file any resource, including resources to which they do not have access.
+        This is not an access control problem because labeling information is private.
+
+        There can only be one folder for a resource. If an attempt is made to set two;
+        the first is overridden with the second.
+        """
+
+        # check for user error
+        if not isinstance(this_resource, BaseResource):
+            raise HSLUsageException("Target is not a resource")
+        label_resource = this_resource.rlabels
+
+        if not isinstance(this_folder, basestring):
+            raise HSLUsageException("Folder is not text")
+
+        # remove leading and trailing whitespace
+        folder_string = UserLabels.clean_folder(this_folder)
+
+        if folder_string == "":
+            raise HSLUsageException("Cannot assign an empty folder name")
```

Comment:
1) Canonical check for empty string is `if not folder_string:` in python.

2) The error message may be confusing to users if the original string they entered was not empty but your cleaning stripped it to being empty. Maybe the error could be more clear?

The exception is documented as not being something you should catch, but this looks like a user error you *do* need to catch. Maybe should just use a standard ValidationError instead?